fsspec/implementations/smb.py

```diff
@@ -0,0 +1,225 @@
+import smbclient
+from stat import S_ISDIR, S_ISLNK
+import types
+import uuid
+from .. import AbstractFileSystem
+from ..utils import infer_storage_options
+
+
+class SMBFileSystem(AbstractFileSystem):
+    """Downloads or uploads to Windows and Samba network drives.
+
+    The argument `path` (str) must have a URI with format:
+    `smb://workgroup;user:password@server:port/share/folder/file.csv`.
```

Comment:
Note that most implementations *accept* complex URLs, and create kwargs from them - as you do - but also allow simple paths, in this case "/share/folder/file.csv", so long as the init kwargs are given.

Reply:
Any suggestion for this?

Reply:
My suggestion is that, in general, "path" arguments can be of the form "/share/folder.file.csv".

Reply:
This is ok according to [this comment bellow](https://github.com/intake/filesystem_spec/pull/340#pullrequestreview-440215111)